src/dynamic_bridge.cpp

```diff
@@ -387,6 +388,9 @@ void get_ros1_service_info(
     return;
   }
   ros::TransportTCPPtr transport(new ros::TransportTCP(nullptr, ros::TransportTCP::SYNCHRONOUS));
+  auto transport_exit = rclcpp::make_scope_exit([&transport]() {
```

Comment:
boost/std shared pointer shouldn't be passed by reference since that undermines the reference counting. Please capture it by value instead which will make sure that the lambda has its own reference which contributes to the count.